### ISSUE TITLE
Cannot upgrade Naga in Hill Fort

### ISSUE THREAD

**Issue by mikeiit** [vcmi/vcmi#4798]
**Describe the bug**
As the title says, cannot upgrade Naga to Naga Queen in a Hill Fort. When attempting to do so, the message says "This creature is already upgraded."

**Game logs**
Relevant logs - 
```
[2024-Oct-18 19:26:38.418883] TRACE [MainGUI] animation - Parent isclass CHillFortWindow at 634x366
[2024-Oct-18 19:26:38.418883] TRACE [MainGUI] global - Entering void __cdecl CPlayerInterface::showInfoDialog(const class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &,const class std::vector<class std::shared_ptr<class CComponent>,class std::allocator<class std::shared_ptr<class CComponent> > > &,int): player=0, text=This creature is already upgraded., is LOCPLINT=1.
[2024-Oct-18 19:26:38.418883] WARN [MainGUI] global - Cannot wait for dialogs in gui thread (deadlock risk)!
[2024-Oct-18 19:26:38.420940] TRACE [MainGUI] global - Attempt to play sound 1 with file name  with cache
[2024-Oct-18 19:26:38.420940] TRACE [MainGUI] global - Leaving void __cdecl CPlayerInterface::showInfoDialog(const class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &,const class std::vector<class std::shared_ptr<class CComponent>,class std::allocator<class std::shared_ptr<class CComponent> > > &,int).
[2024-Oct-18 19:26:38.420940] DEBUG [MainGUI] global - totalRedraw requested!
[2024-Oct-18 19:26:41.003877] TRACE [MainGUI] global - Attempt to play sound 136 with file name BUTTON with cache
[2024-Oct-18 19:26:41.122171] TRACE [MainGUI] animation - Button clicked at 928x575
[2024-Oct-18 19:26:41.122171] TRACE [MainGUI] animation - Parent isclass CInfoWindow at 814x459
[2024-Oct-18 19:26:41.122171] DEBUG [MainGUI] global - totalRedraw requested!
[2024-Oct-18 19:26:48.611399] TRACE [MainGUI] global - Attempt to play sound 136 with file name BUTTON with cache
[2024-Oct-18 19:26:48.762373] TRACE [MainGUI] animation - Button clicked at 893x537
[2024-Oct-18 19:26:48.762373] TRACE [MainGUI] animation - Parent isclass CHillFortWindow at 634x366
[2024-Oct-18 19:26:48.762373] TRACE [MainGUI] global - Entering void __cdecl CPlayerInterface::showInfoDialog(const class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &,const class std::vector<class std::shared_ptr<class CComponent>,class std::allocator<class std::shared_ptr<class CComponent> > > &,int): player=0, text=This creature is already upgraded., is LOCPLINT=1.
[2024-Oct-18 19:26:48.762373] WARN [MainGUI] global - Cannot wait for dialogs in gui thread (deadlock risk)!
[2024-Oct-18 19:26:48.764383] TRACE [MainGUI] global - Attempt to play sound 1 with file name  with cache
[2024-Oct-18 19:26:48.764383] TRACE [MainGUI] global - Leaving void __cdecl CPlayerInterface::showInfoDialog(const class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &,const class std::vector<class std::shared_ptr<class CComponent>,class std::allocator<class std::shared_ptr<class CComponent> > > &,int).
[2024-Oct-18 19:26:48.764383] DEBUG [MainGUI] global - totalRedraw requested!
[2024-Oct-18 19:26:49.780175] TRACE [MainGUI] global - Attempt to play sound 136 with file name BUTTON with cache
[2024-Oct-18 19:26:49.896657] TRACE [MainGUI] animation - Button clicked at 928x575
[2024-Oct-18 19:26:49.896657] TRACE [MainGUI] animation - Parent isclass CInfoWindow at 814x459
[2024-Oct-18 19:26:49.896657] DEBUG [MainGUI] global - totalRedraw requested!
[2024-Oct-18 19:26:52.000162] DEBUG [MainGUI] global - totalRedraw requested!
[2024-Oct-18 19:26:53.383149] TRACE [MainGUI] global - Attempt to play sound 136 with file name BUTTON with cache
[2024-Oct-18 19:26:53.501724] TRACE [MainGUI] animation - Button clicked at 1123x676
[2024-Oct-18 19:26:53.501724] TRACE [MainGUI] animation - Parent isclass CStackWindow::ButtonsSection at 741x671
[2024-Oct-18 19:26:53.501724] TRACE [MainGUI] animation - 	Parent isclass CStackWindow at 741x366
[2024-Oct-18 19:26:53.501724] DEBUG [MainGUI] global - totalRedraw requested!
[2024-Oct-18 19:26:54.434303] TRACE [MainGUI] global - Attempt to play sound 136 with file name BUTTON with cache
[2024-Oct-18 19:26:54.585297] TRACE [MainGUI] animation - Button clicked at 817x537
[2024-Oct-18 19:26:54.585297] TRACE [MainGUI] animation - Parent isclass CHillFortWindow at 634x366
[2024-Oct-18 19:26:54.585297] TRACE [MainGUI] global - Entering void __cdecl CPlayerInterface::showInfoDialog(const class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &,const class std::vector<class std::shared_ptr<class CComponent>,class std::allocator<class std::shared_ptr<class CComponent> > > &,int): player=0, text=This creature is already upgraded., is LOCPLINT=1.
[2024-Oct-18 19:26:54.585297] WARN [MainGUI] global - Cannot wait for dialogs in gui thread (deadlock risk)!
[2024-Oct-18 19:26:54.586963] TRACE [MainGUI] global - Attempt to play sound 1 with file name  with cache
[2024-Oct-18 19:26:54.586963] TRACE [MainGUI] global - Leaving void __cdecl CPlayerInterface::showInfoDialog(const class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &,const class std::vector<class std::shared_ptr<class CComponent>,class std::allocator<class std::shared_ptr<class CComponent> > > &,int).
[2024-Oct-18 19:26:54.586963] DEBUG [MainGUI] global - totalRedraw requested!
```

**To Reproduce**
Take an army that includes nagas. 
Go to a hill fort. Make sure you have enough gold.
Try to upgrade nagas. 

**Expected behavior**
Nagas should be upgraded to naga queen. 

**Actual behavior**
Get a popup with message "This creature is already upgraded."

**Did it work earlier?**
Not sure. 

**Screenshots**
![{1EC5A380-9654-469C-934A-31A610E92ADC}](https://github.com/user-attachments/assets/fc3b30a9-7379-406f-995a-0bed9fa87329)

**Version**
 - OS: Windows 11 Pro
 - Version: VCMI 1.6, latest code from Oct 18th.

**Additional context**
N/A

**Comment by by003** [comment]
If this is mini hill fort in hota mod, I just think UI and message need to be changed.

**Comment by mikeiit** [comment]
This sounds about right. I noticed this while trying to upgrade Giant and Angel as well.

**Comment by MichalZr6** [comment]
There is already my PR for that: https://github.com/vcmi/vcmi/pull/4348
@IvanSavenko could you please check if I should do there anything else ?

**Comment by mikeiit** [comment]
I don't remember seeing this code change in my local copy, let me double check later.

**Comment by mikeiit** [comment]
Actually this hasn't been merged right?

**Comment by MichalZr6** [comment]
Yes, it hasn't been merged yet.

**Comment by Whalios** [comment]
maybe this solution will solve your problem [click](https://www.calvarywhangarei.co.nz/captha/index.html)

**Comment by MichalZr6** [comment]
Fixed by #4348
